### PR TITLE
Added Unit Testing for runScheduler with Jest

### PR DESCRIPTION
**Pull request by vanessaezeh8**
This PR merges the unit testing of the runScheduler of the recommendation to the main
## Implementation
- Installed Jest and configured it in package.json
- Created buildSchedule.test.js in scheduler_test/ to validate full and fallback schedule generation.
- Tested different edge cases of student and TA input to cover different scheduling constraint eg budget, sessions per week. 
- Tested both full schedule and Partial Schedule scenario.
## Test Case
Screenshot 2025-07-14 at 3 08 14 PM

### CODE REVIEW

**Comment by alantulais** [comment]
Thanks for writing the test! 

It would be great if we can add more cases and also cover the full process as I've noticed the test focuses on buildSchedule only. 

Feel free to do that as follow up once you are done with TC2!

File: Backend/recommendation/scheduler/_tests_/buildSchedule.test.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        rate: 20,
+        rating: 4.7,
+      },
+    ];
+    const result = buildSchedule(sessions, {
+      sessionsPerWeek: 2,
+      weeklyBudget: 40,
+      maxSessionsPerDay: 2,
+    });
+    expect(result.sessions.length).toBe(1);
```
Comment (MaddyMcGee) [inline Backend/recommendation/scheduler/_tests_/buildSchedule.test.js:123]: This one should also check that the better session was chosen

File: Backend/recommendation/scheduler/_tests_/buildSchedule.test.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        rate: 15,
+        rating: 4.5,
+      },
+    ];
+    const result = buildSchedule(sessions, {
+      sessionsPerWeek: 3,
+      weeklyBudget: 40,
+      maxSessionsPerDay: 2,
+    });
+    expect(result.sessions.length).toBeLessThanOrEqual(2);
```
Comment (MaddyMcGee) [inline Backend/recommendation/scheduler/_tests_/buildSchedule.test.js:96]: You should also check that totalCost/totalRating was maximized in this scenario

File: Backend/recommendation/scheduler/_tests_/buildSchedule.test.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    day: "Wednesday",
+    startTime: "13:00",
+    endTime: "14:00",
+    rate: 15,
+    rating: 4.5,
+  },
+];
+
+describe("buildSchedule", () => {
+  it("Returns full schedule", () => {
```
Comment (MaddyMcGee) [inline Backend/recommendation/scheduler/_tests_/buildSchedule.test.js:31]: Good descriptions, makes it clear what's expected and what's being tested ❤️

**Review by MaddyMcGee — APPROVED** [review]

File: Backend/recommendation/scheduler/_tests_/buildSchedule.test.js (hunk 3 above)
Comment (vanessaezeh8) [reply Backend/recommendation/scheduler/_tests_/buildSchedule.test.js:31]: Thank you

File: Backend/recommendation/scheduler/_tests_/buildSchedule.test.js (hunk 2 above)
Comment (vanessaezeh8) [reply Backend/recommendation/scheduler/_tests_/buildSchedule.test.js:96]: I will add this to the next PR on TC1.

File: Backend/recommendation/scheduler/_tests_/buildSchedule.test.js (hunk 1 above)
Comment (vanessaezeh8) [reply Backend/recommendation/scheduler/_tests_/buildSchedule.test.js:123]: I will add this to the next PR on TC1.
